lightning/src/ln/channelmanager.rs

```diff
@@ -1765,6 +1813,21 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 			return Err(APIError::APIMisuseError { err: format!("Channel value must be at least 1000 satoshis. It was {}", channel_value_satoshis) });
 		}
 
+		let height = self.best_block.read().unwrap().height();
+		let mut outbound_scid_alias = 0;
+		let mut i = 0;
+		loop {
+			#[cfg(fuzzing)] { // fuzzing chacha20 doesn't use the key at all so we always get the same alias
+				outbound_scid_alias += 1;
+			}
+			#[cfg(not(fuzzing))] {
+				outbound_scid_alias = fake_scid::Namespace::OutboundAlias.get_fake_scid(height, &self.genesis_hash, &self.fake_scid_rand_bytes, &self.keys_manager);
+			}
+			if self.outbound_scid_aliases.lock().unwrap().insert(outbound_scid_alias) { break; }
+			i += 1;
+			if i > 1_000_000 { panic!("Your RNG is busted or we ran out of possible outbound SCID aliases (which should never happen before we run out of memory to store channels"); }
```

Comment:
Should this be done in `internal_open_channel`, too (in a DRY manner)?

Reply:
Yea, this got added later and the thing keeps growing, I'll DRY it.